### PR TITLE
Allow export of raw anonymize data

### PR DESCRIPTION
**Pull request by vkryukov**
Suggestion for creating an anonymized raw data export, consisting of timestamps, hash of codes, and actual votes.

### CODE REVIEW

File: elections57_results.py [hunk 1]
```diff
@@ -220,6 +228,23 @@ def get_class(v):
...
+    # Symbolic names for spreadsheet columns:
+    col_timestamp, col_has_code, col_code, col_name, col_class, \
+    col_fbpage, col_vkpage, col_email, col_sub_news, col_pub_dir, \
+    col_candidates, col_bylaws, col_comment = range(13)     
+
+    with open(OUTPUT, 'wb') as f:
+        writer = csv.writer(f)
+        writer.writerow(['Time', 'Hash', 'Votes'])
+        for row in raw_data:
+            writer.writerow([row[col_timestamp], hash(row[col_code]), row[col_candidates]])
```
Comment (vadz) [inline elections57_results.py:247]: I think you need `.encode(utf-8)` for the last argument here.

File: elections57_results.py [hunk 2]
```diff
@@ -220,6 +228,23 @@ def get_class(v):
         for c in sorted(classes):
             votes_by_class.append(votes[c])
         print '{},{}'.format(y, ','.join([str(x) for x in votes_by_class]))
+elif cmdline_args.operation == 'raw_data':
+    SALT = input("Enter salt: ") # Or any other method to get SALT known to organizers
```
Comment (vadz) [inline elections57_results.py:232]: I'd take it on command line, as an option -- script runs should be reproducible,

File: elections57_results.py [hunk 3]
```diff
@@ -220,6 +228,23 @@ def get_class(v):
...
+    OUTPUT = 'raw_results_anonymized.csv'
+
+    def hash(code):
+        return hashlib.sha256(code + SALT).hexdigest()
```
Comment (vadz) [inline elections57_results.py:236]: I _think_ this is secure enough, but I'm _not_ 100% sure that this is safe if somebody really bothers to precompute a "rainbow table" with enough hashes and all voters names (see below). I'd use bcrypt just to be totally safe.

**Review by vadz — CHANGES_REQUESTED** [review]
You need to deal with people who voted without codes, i.e. compute the hash from the class-name-surname if the code is empty too.

**Comment by vkryukov** [comment]
Closing this to make the new code based on Vadim's other suggestion

**Pull request closed without merging**